docs/source/config/server.rst

```diff
@@ -0,0 +1,214 @@
+
+``PERMANENT_SESSION_LIFETIME``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Session cookie's lifetime.
+This is the equivalent of Flask's `PERMANENT_SESSION_LIFETIME <https://flask.palletsprojects.com/en/2.3.x/config/#PERMANENT_SESSION_LIFETIME>`_  configuration value.
+
+``SECURITY_PASSWORD_SALT``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Password hashing salt.
+This is the equivalent of Flask-Security-Too's `SECURITY_PASSWORD_SALT <https://flask-security-too.readthedocs.io/en/stable/configuration.html#SECURITY_PASSWORD_SALT>`_  configuration value.
+
+``SECURITY_TRACKABLE``
+~~~~~~~~~~~~~~~~~~~~~~
+
+Enables tracking of basic user login statistics.
+This is the equivalent of Flask-Security-Too's `SECURITY_TRACKABLE <https://flask-security-too.readthedocs.io/en/stable/configuration.html#SECURITY_TRACKABLE>`_  configuration value.
+
+Default value is ``True``.
+
+``USER_DB_PATH``
+~~~~~~~~~~~~~~~~
+Full path to SQLite database file, used for storing users and sessions data.
+
+Default value is ``/tmp/mxcube-user.db``.
+
+``ALLOWED_CORS_ORIGINS``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+List of origins that are allowed to connect to this server.
+This list is passed via ``cors_allowed_origins`` parameter to the `SocketIO <https://flask-socketio.readthedocs.io/en/latest/api.html#flask_socketio.SocketIO>`_ class constructor.
+Note that if this list is too restrictive, the front-end will fail to create a SocketIO connection to the back-end.
+
+``CERT``
+~~~~~~~~
+
+Used to enable SSL and set the type of host certificate.
+The following values are supported:
+
++--------+------------------------------------------------+
+| NONE   | SSL disabled                                   |
++--------+------------------------------------------------+
+| ADHOC  | SSL enabled, autogenerated certificate is used |
++--------+------------------------------------------------+
+| SIGNED | SSL enabled, signed certificate is used        |
++--------+------------------------------------------------+
+
+Default value is ``NONE``.
+
+``CERT_PEM``
+~~~~~~~~~~~~
+
+Path of the certificate file to use for SSL connections.
+
+``CERT_KEY``
+~~~~~~~~~~~~
+
+Path of the private key file to use for SSL connections.
+
+mxcube
+------
+
+The mxcube section allows you to customize miscellaneous aspects of the MxCube application.
+It have settings for the on-axis video stream, user account management, experiment mode.
+The section has the following syntax:
+
+.. code-block:: yaml
+
+    mxcube:
+      mode: <experiment mode>
+      USE_EXTERNAL_STREAMER: <boolean>
+      VIDEO_FORMAT: <format name>
+      VIDEO_STREAM_URL: <stream URL>
+      VIDEO_STREAM_PORT: <stream port>
+      usermanager:
+        class: <class name>
+        inhouse_is_staff: <boolean>
+        users:
+          - username: <user name>
+            role: <role name>
+          ...
+
+The ``mxcube`` section may contain keys as described below.
+
+``mode``
```

Comment:
Unrelated to the documentation, but wondering if this should be changed to `PASCAL_CASE` for consistency (same with other `snake_case` keys).

Reply:
You mean the mode names? Like:

    SSX-CHIP -> SSX_CHIP
    SSX-INJECTOR -> SSX_INJECTOR

yes, I agree.

Reply:
Sorry I meant the key itself, `MODE`, for consistency with the other keys.

... unless I've missed that the distinction matters for some reason? Are `PASCAL_CASE` keys ones that can be overridden via environment variables or something? I don't necessarily think that keys in a YAML file should be written in `PASCAL_CASE` (unlike keys in a `.env` file for instance), but it should definitely be consistent (or the reason why it's not should perhaps be documented).

Reply:
_[Aside: @axelboc, this is [`PascalCase`](https://en.wikipedia.org/wiki/Camel_case), isn't it? You probably mean [`SCREAMING_SNAKE_CASE`](https://en.wikipedia.org/wiki/Snake_case), no?]_

Reply:
_[@axelboc, the names are silly, and I always have to look them up anyway. Seeing `CONSTANT_CASE` gets the message across much better for me, indeed.]_

Reply:
I agree that we should make the casing consistent in server.yaml.

But think that's a breaking change, so needs to be carefully managed. In other words this is a topic for another PR :)